include/LLVMSPIRVLib.h

```diff
@@ -106,6 +106,21 @@ std::unique_ptr<SPIRVModule> readSpirvModule(std::istream &IS,
                                              const SPIRV::TranslatorOpts &Opts,
                                              std::string &ErrMsg);
 
+/// \brief Partially load SPIR-V from the stream and decode only selected
+/// for the report instructions, needed to retrieve general information
+/// about the module.
+/// \returns nullopt on failure.
```

Comment:
Should this be moved below to be next to a function instead of a struct? Should we note that if this API returned failure, then `readSPIRVModule` is expected to fail as well?

Reply:
Agree, thanks